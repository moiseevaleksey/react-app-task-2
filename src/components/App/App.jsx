import React from 'react';
import { Switch, Route, withRouter } from 'react-router';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import style from 'stylesheet/main.scss';

import Header from '../Header/Header';
import Main from '../Main/Main';
import Login from '../Login/Login';
import Registration from '../Registration/Registration';
import { checkUser } from '../../actions/user';

export class App extends React.Component {
  static propTypes = {
    checkUser: PropTypes.func,
  };

  componentDidMount() {
    this.props.checkUser();
  }

  render() {
    return (
      <div className={style.app}>
        <Header />
        <Switch>
          <Route exact path="/" component={Main} />
          <Route path="/login" component={Login} />
          <Route path="/register" component={Registration} />
        </Switch>
      </div>
    );
  }
}

export default withRouter(connect(null, { checkUser })(App));
