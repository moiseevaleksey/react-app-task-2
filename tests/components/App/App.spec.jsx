import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { App } from './../../../src/components/App/App';

describe('App', () => {
  let wrapper;

  it('App should be rendered with props', () => {
    wrapper = shallow(<App checkUser={()=>{}}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
