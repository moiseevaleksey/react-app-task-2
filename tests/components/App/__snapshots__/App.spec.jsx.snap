// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should be rendered with props 1`] = `
<div
  className="app"
>
  <Connect(Header) />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/login"
    />
    <Route
      component={[Function]}
      path="/register"
    />
  </Switch>
</div>
`;
